opslevel/datasource_opslevel_webhook_action.go

```diff
@@ -1,72 +1,125 @@
-// 				Description: "The name of the Webhook Action.",
-// 				Computed:    true,
-// 			},
-// 			"description": {
-// 				Type:        schema.TypeString,
-// 				Description: "The description of the Webhook Action.",
-// 				Computed:    true,
-// 			},
-// 			"payload": {
-// 				Type:        schema.TypeString,
-// 				Description: "Template that can be used to generate a webhook payload.",
-// 				Computed:    true,
-// 			},
-// 			"url": {
-// 				Type:        schema.TypeString,
-// 				Description: "The URL of the Webhook Action.",
-// 				Computed:    true,
-// 			},
-// 			"method": {
-// 				Type:        schema.TypeString,
-// 				Description: "The http method used to call the Webhook Action.",
-// 				Computed:    true,
-// 			},
-// 			"headers": {
-// 				Type:        schema.TypeMap,
-// 				Description: "HTTP headers to be passed along with your webhook when triggered.",
-// 				Computed:    true,
-// 				Elem: &schema.Schema{
-// 					Type: schema.TypeString,
-// 				},
-// 			},
-// 		},
-// 	}
-// }
-
-// func datasourceWebhookActionRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	identifier := d.Get("identifier").(string)
-// 	resource, err := client.GetCustomAction(identifier)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	d.SetId(string(resource.Id))
-// 	d.Set("name", resource.Name)
-// 	d.Set("description", resource.Description)
-// 	d.Set("payload", resource.LiquidTemplate)
-// 	d.Set("url", resource.WebhookURL)
-// 	d.Set("method", resource.HTTPMethod)
-// 	d.Set("headers", resource.Headers)
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure WebhookActionDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &WebhookActionDataSource{}
+
+func NewWebhookActionDataSource() datasource.DataSource {
+	return &WebhookActionDataSource{}
+}
+
+// WebhookActionDataSource manages a WebhookAction data source.
+type WebhookActionDataSource struct {
+	CommonDataSourceClient
+}
+
+// WebhookActionDataSourceModel describes the data source data model.
+type WebhookActionDataSourceModel struct {
+	Description types.String `tfsdk:"description"`
+	Headers     types.Map    `tfsdk:"headers"`
+	Identifier  types.String `tfsdk:"identifier"`
+	Method      types.String `tfsdk:"method"`
+	Name        types.String `tfsdk:"name"`
+	Payload     types.String `tfsdk:"payload"`
+	Url         types.String `tfsdk:"url"`
+}
+
+func jsonToMap(json map[string]any) map[string]attr.Value {
+	jsonAttrs := make(map[string]attr.Value)
+	for k, v := range json {
+		jsonAttrs[k] = types.StringValue(v.(string))
```

Comment:
Wouldn't this nil panic?

Reply:
Yes this could go horribly wrong. 😁 
I got this working working and never came back to it

Reply:
To be precise, this would cause a `panic: interface conversion: ...`